src/Config/Level/CodeQualityLevel.php

```diff
@@ -0,0 +1,213 @@
+use Rector\CodeQuality\Rector\If_\ShortenElseIfRector;
+use Rector\CodeQuality\Rector\If_\SimplifyIfElseToTernaryRector;
+use Rector\CodeQuality\Rector\If_\SimplifyIfNotNullReturnRector;
+use Rector\CodeQuality\Rector\If_\SimplifyIfNullableReturnRector;
+use Rector\CodeQuality\Rector\If_\SimplifyIfReturnBoolRector;
+use Rector\CodeQuality\Rector\Include_\AbsolutizeRequireAndIncludePathRector;
+use Rector\CodeQuality\Rector\Isset_\IssetOnPropertyObjectToPropertyExistsRector;
+use Rector\CodeQuality\Rector\LogicalAnd\AndAssignsToSeparateLinesRector;
+use Rector\CodeQuality\Rector\LogicalAnd\LogicalToBooleanRector;
+use Rector\CodeQuality\Rector\New_\NewStaticToNewSelfRector;
+use Rector\CodeQuality\Rector\NotEqual\CommonNotEqualRector;
+use Rector\CodeQuality\Rector\NullsafeMethodCall\CleanupUnneededNullsafeOperatorRector;
+use Rector\CodeQuality\Rector\Switch_\SingularSwitchToIfRector;
+use Rector\CodeQuality\Rector\Switch_\SwitchTrueToIfRector;
+use Rector\CodeQuality\Rector\Ternary\ArrayKeyExistsTernaryThenValueToCoalescingRector;
+use Rector\CodeQuality\Rector\Ternary\NumberCompareToMaxFuncCallRector;
+use Rector\CodeQuality\Rector\Ternary\SimplifyTautologyTernaryRector;
+use Rector\CodeQuality\Rector\Ternary\SwitchNegatedTernaryRector;
+use Rector\CodeQuality\Rector\Ternary\TernaryEmptyArrayArrayDimFetchToCoalesceRector;
+use Rector\CodeQuality\Rector\Ternary\UnnecessaryTernaryExpressionRector;
+use Rector\CodingStyle\Rector\ClassMethod\FuncGetArgsToVariadicParamRector;
+use Rector\CodingStyle\Rector\FuncCall\CallUserFuncToMethodCallRector;
+use Rector\CodingStyle\Rector\FuncCall\CountArrayToEmptyArrayComparisonRector;
+use Rector\Contract\Rector\RectorInterface;
+use Rector\Php52\Rector\Property\VarToPublicPropertyRector;
+use Rector\Php71\Rector\FuncCall\RemoveExtraParametersRector;
+use Rector\Renaming\Rector\FuncCall\RenameFunctionRector;
+use Rector\Strict\Rector\Empty_\DisallowedEmptyRuleFixerRector;
+
+/**
+ * Key 0 = level 0
+ * Key 50 = level 50
+ *
+ * Start at 0, go slowly higher, one level per PR, and improve your rule coverage
+ *
+ * From the safest rules to more changing ones.
+ *
+ * @experimental This list can change in time, based on community feedback,
+ * what rules are safer than other. The safest rules will be always in the top.
+ */
+final class CodeQualityLevel
+{
+    /**
+     * The rule order matters, as its used in withCodeQualityLevel() method
+     * Place the safest rules first, follow by more complex ones
+     *
+     * @var array<class-string<RectorInterface>>
+     */
+    public const RULES = [
+        CombinedAssignRector::class,
+        SimplifyEmptyArrayCheckRector::class,
+        ReplaceMultipleBooleanNotRector::class,
+        ForeachToInArrayRector::class,
+        SimplifyForeachToCoalescingRector::class,
+        SimplifyFuncGetArgsCountRector::class,
+        SimplifyInArrayValuesRector::class,
+        SimplifyStrposLowerRector::class,
+        GetClassToInstanceOfRector::class,
```

Comment:
This can be moved to last due to possible unresolvable issue:

- https://github.com/rectorphp/rector/issues/8639

That maybe deprecated/removed from the list in the future.